src/Result/Api/GeneralPurposeResultHandler.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: olsiqose
+ * Date: 09/07/2019
+ * Time: 11.51
+ */
+
+namespace Nosto\Result\Api;
+
+use Nosto\Request\Http\HttpResponse;
+
+final class GeneralPurposeResultHandler extends ApiResultHandler
+{
+    /**
+     * @return GeneralPurposeResultHandler|null
+     */
+    public static function getInstance()
+    {
+        static $inst = null;
```

Comment:
How does this work -  can you really have a static variable definition inside the function?